docs/modules/ROOT/pages/Java_Intro_to_Mill.adoc

```diff
@@ -0,0 +1,84 @@
+// plenty of space to go in-depth.
+//
+// By the end of this page, a prospective Mill user should be familiar with
+// what Mill is, hopefully have downloaded an example to try out, and be
+// interested in learning more about the Mill build tool
+
+= Introduction to Mill for Java
+
+:page-aliases: index.adoc, Intro_to_Mill.adoc, Intro_to_Mill_for_Java.adoc
+
+{mill-github-url}[Mill] is a Java build tool that aims for performance,
+maintainability, and flexibility:
+
+:language: Java
+
+include::partial$Intro_to_Mill_Header.adoc[]
+
+Mill borrows ideas from other tools like https://maven.apache.org/[Maven],
+https://gradle.org/[Gradle], https://bazel.build/[Bazel], but tries to learn from the
+strengths of each tool and improve on their weaknesses. Although Maven and Gradle
+are mature widely-used tools, they have fundamental limitations in their design
+(https://blog.ltgt.net/maven-is-broken-by-design/[Maven Design],
+https://www.bruceeckel.com/2021/01/02/the-problem-with-gradle/[Gradle Design]) that make
+them difficult to improve upon incrementally. Mill tries to do much better.
+
+Compared to Maven:
+
+* **Mill follows Maven's innovation of good built-in defaults**: Mill's built-in
+  ``JavaModule``s follow Maven's "convention over configuration" style. This provides an
+  excellent default template that can be used to structure your project without any custom
+  configuration.
+
+* **Mill makes customizing the build tool much easier than Maven**. As projects grow, build
+  grow beyond just compiling a single language: you need custom
+  code generation, custom linting workflows, custom output artifacts, or support for
+  additional programming languages. Mill makes doing this yourself easy, so you are not beholden
+  to third-party plugins that may not exist, be well maintained, or interact well with each other.
+
+* **Mill automatically caches and parallelizes your build**: This applies not just to the
+  built-in ``JavaModule``s and tasks that Mill ships with, but also any custom logic the
+  developer may configure in their Mill build. This maximizes performance and snappiness of
+  your command-line build workflows, and especially matters in larger codebases where builds
+  tend to get slow.
+
+Compared to Gradle:
+
+* **Mill follows Gradle's conciseness**: Rather than pages and pages of verbose XML, every
+  line of configuration in a Mill build is meaningful. e.g. adding a dependency is 1 line in
+  Mill, like it is in Gradle, and unlike the 5 line `<dependency>` declaration you find in Maven
+
+* **Mill helps keep your build understandable**: Unlike Gradle config statements that can
+  work in mysterious ways, Mill build configuration is done via vanilla methods, objects,
+  inheritance, and overrides. This means that impact of a Mill config statement is localized,
+  without "_spooky action at a distance_", and the exact meaning of any config statement is
+  a single "_jump-to-definition_" away in your IDE.
+
+
```

Comment:
@lefou @lolgab do you guys use Gradle? I'm trying to come up with a third bullet point to put here, but not sure what it should be. What are the things a gradle user would find most interesting about Mill?